src/main/kotlin/com/cognifide/gradle/aem/instance/InstanceSync.kt

```diff
@@ -367,19 +196,17 @@ class InstanceSync(val project: Project, val instance: Instance) {
 
         logger.info("Installing package $remotePath on $instance")
 
-        val json = try {
-            postMultipart(url, mapOf("recursive" to config.installRecursive))
-        } catch (e: Exception) {
-            throw DeployException("Cannot install package $remotePath on $instance. Reason: request failed.", e)
-        }
-
         val response = try {
-            InstallResponse(json)
-        } catch (e: Exception) {
-            throw DeployException("Malformed install response after installing package $remotePath on $instance.", e)
-        }
+            postMultipart(url, mapOf("recursive" to config.installRecursive)) { InstallResponse.from(asStream(it), config.packageErrors) }
+        } catch (e: RequestException) {
 
-        if (!response.success) {
+            throw DeployException("Cannot install package $remotePath on $instance. Reason: request failed.", e)
+        } catch (e: ResponseException) {
+            throw DeployException("Malformed response after installing package $remotePath on $instance.")
+        }
+        if (response.hasPackageErrors) {
+            throw MalformedPackageException.of(response, "Encountered critical installation error(s) in package $remotePath:")
```

Comment:
the `MalformedPackageException`  is too precise for me.  `PackageException` / we would like only indicate that the reason of exception is package itself so that "malformed" could be just a part of msg. w.g if we decide to throw PackageException or `MissingPackageException` we will have a base in our hierarchy already which will be `PackageException` .